pkg/controller/baremetalhost/host_state_machine.go

```diff
@@ -152,13 +153,24 @@ func (hsm *hostStateMachine) shouldInitiateRegister(info *reconcileInfo) bool {
 }
 
 func (hsm *hostStateMachine) handleNone(info *reconcileInfo) actionResult {
-	// Running the state machine at all means we have successfully validated
-	// the BMC credentials once, so we can move to the Registering state.
-	hsm.Host.ClearError()
-	hsm.NextState = metal3v1alpha1.StateRegistering
+	// No state is set, so immediately move to either Registering or Unmanaged
+	if hsm.Host.HasBMCDetails() {
+		hsm.NextState = metal3v1alpha1.StateRegistering
+	} else {
+		hsm.Host.SetOperationalStatus(metal3v1alpha1.OperationalStatusDiscovered)
+		hsm.NextState = metal3v1alpha1.StateUnmanaged
+	}
 	return actionComplete{}
 }
 
+func (hsm *hostStateMachine) handleUnmanaged(info *reconcileInfo) actionResult {
+	if hsm.Host.HasBMCDetails() {
+		hsm.NextState = metal3v1alpha1.StateRegistering
+		return actionComplete{}
+	}
+	return actionContinue{}
```

Comment:
This is wrong, we don't actually need to reschedule at all here (we won't leave this state unless the CR is modified) - or at the very least not until after a long delay. This will spin in tight loop which is very bad.